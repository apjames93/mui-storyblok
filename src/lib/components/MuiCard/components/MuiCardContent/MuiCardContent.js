import React from 'react';
import PropTypes from 'prop-types';
import CardContent from '@material-ui/core/CardContent';
import { validComponents } from '../../../../utils/customProps';
import Storyblok from '../../../../utils/Storyblok';
import MuiTypography from '../../../MuiTypography/MuiTypography';
import MuiAccordion from '../../../MuiAccordion/MuiAccordion';
import MuiList from '../../../MuiList/MuiList';
import { renderComponents } from '../../../../utils/customComponents';

const MuiCardContent = ({
  content,
  rootClass,
}) => {
  const components = {
    MuiTypography,
    MuiAccordion,
    MuiList,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <CardContent
      className={styles.root}
    >
      {content.map((component, key) => renderComponents(components, component, key))}
    </CardContent>
  );
};

export default MuiCardContent;

MuiCardContent.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * Content passed to render
   * components:
    MuiTypography,
    MuiAccordion,
    MuiList,
   */
  content(props, propName, componentName) {
    const components = [
      'MuiTypography',
      'MuiAccordion',
      'MuiList',
    ];
    return validComponents(props, propName, componentName, components);
  },
};

MuiCardContent.defaultProps = {
  rootClass: [],
  content: [],
};
