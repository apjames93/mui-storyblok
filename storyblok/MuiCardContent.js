module.exports = {
  "name": "MuiCardContent",
  "display_name": null,
  "updated_at": "2020-07-20T17:00:14.646Z",
  "schema": {
    "rootClass": {
      "type": "options",
      "description": "Mui Override or extend the styles applied to the component."
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiTypography",
        "MuiList",
        "MuiAccordion"
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiCardContent",
  "component_group_uuid": null
}