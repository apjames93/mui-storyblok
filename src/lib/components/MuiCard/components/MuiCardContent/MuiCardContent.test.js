import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiCardContent from './MuiCardContent';

function setup() {
  const props = {
    content: [{
      component: 'MuiTypography',
      content: [{
        component: 'MuiText',
        text: 'text',
      }],
    }, {
      component: 'MuiExpansionPanel',
      expansionPanelSummary: [{
        component: 'MuiExpansionPanelSummary',
        content: [{
          component: 'MuiIcon',
          iconName: 'android',
        }],
        expandIcon: [{
          component: 'MuiIcon',
          iconName: 'android',
        }],
      }],
      expansionPanelDetails: [{
        component: 'MuiExpansionPanelDetails',
        content: [{
          component: 'MuiExpansionPanelTypography',
          content: [{
            component: 'MuiText',
            text: 'text',
          }],
        }],
      }],
    }],
  };
  const comp = shallow(<MuiCardContent {...props} />);
  return { comp, props };
}

describe('<MuiCardContent />', () => {
  it('renders MuiCardContent', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create((
      <MemoryRouter>
        <MuiCardContent {...props} />
      </MemoryRouter>
    ));
    expect(tree).toMatchSnapshot();
  });
});
