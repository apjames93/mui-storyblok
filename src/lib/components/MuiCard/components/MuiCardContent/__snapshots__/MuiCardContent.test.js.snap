// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiCardContent /> snapshot 1`] = `
<div
  className="MuiCardContent-root makeStyles-root-3"
>
  <div
    className="MuiTypography-root makeStyles-root-4 MuiTypography-body1"
  >
    <span
      className="MuiTypography-root makeStyles-root-5 MuiTypography-body1"
    >
       undefined 
    </span>
  </div>
  <div
    className="MuiPaper-root MuiAccordion-root makeStyles-root-6 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-root-7"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content"
      >
        <span
          aria-hidden={true}
          className="material-icons MuiIcon-root makeStyles-root-8 MuiIcon-colorSecondary"
        >
          android
        </span>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root makeStyles-root-9 MuiIcon-colorSecondary"
          >
            android
          </span>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root makeStyles-root-10"
            >
              <div
                className="MuiTypography-root makeStyles-root-11 MuiTypography-body1"
              >
                <span
                  className="MuiTypography-root makeStyles-root-12 MuiTypography-body1"
                >
                   undefined 
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
